autoscheduler/frontend/src/components/Navbar/LoginButton.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { Button } from '@material-ui/core';
+import ExitToAppOutlinedIcon from '@material-ui/icons/ExitToAppOutlined';
+// This file contains the login button. When a user logs in, it is replaced by their name
+// followed by the logout button.
+const LoginButton: React.FC = () => {
+  // Used to track whether the API call has returned anything yet
+  const [loading, setLoading] = React.useState(true);
+
+  // used to logout the user
+  function logout(): void {
+    fetch('sessions/logout').then(
+      (res) => {
+        if (res.ok) {
+          window.location.reload();
+          return;
+        }
+        throw new Error(res.status.toString());
+      },
+    ).catch(() => { });
+  }
+
+  // This checks for the logged in user's name and sets it if found.
+  // Otherwise, it throws an error and catches it so nothing breaks.
+  const [usersName, setUsersName] = React.useState('');
+  React.useEffect(() => {
+    fetch('sessions/get_full_name').then(
+      (res) => {
+        if (res.ok) {
+          return res.json();
+        }
+
+        throw new Error(res.status.toString());
+      },
+    ).then(({ fullName }) => {
+      if (fullName) setUsersName(fullName);
+    }).catch(() => { })
+      .finally(() => { setLoading(false); });
+  }, []);
+
+  // Checks to make sure fetch call has finished before showing anything
+  if (loading) {
```

Comment:
Could simplify to one line

Reply:
do you mean like `if (loading) return null;` ?

Reply:
If so, I am aware. I just left it the way I did because I like having the curly braces. I can change it no problem though if you want.